src/test/java/org/jvnet/hudson/test/JenkinsRuleTest.java

```diff
@@ -92,4 +155,27 @@ public String getSetterParam() {
             return setterParam;
         }
     }
+
+    @TestExtension("testTokenHelperMethods")
+    public static class AuthRetrieval implements UnprotectedRootAction {
+        @Override
+        public String getIconFileName() {
+            return null;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return null;
+        }
+
+        @Override
+        public String getUrlName() {
+            return "test";
+        }
+
+        public HttpResponse doIndex() {
+            User u = User.current();
+            return HttpResponses.plainText(u!=null ? u.getId() : "anonymous");
```

Comment:
BTW there is also a standard `/whoAmI` though the response is HTML so it is probably less convenient for purposes of this test. Or you can use `/me/api/xml?xpath=/*/id/text()`.

Reply:
The usage of XPath in the url throw me a 403, it requires SecureRequester or something like that. I didn't investigate more, I just used `whoAmI/api/json` and `JSONObject` to do the job.